src/main/scala/pipelines/text/NewsgroupsPipeline.scala

```diff
@@ -21,15 +21,17 @@ object NewsgroupsPipeline extends Logging {
 
     // Build the classifier estimator
     logInfo("Training classifier")
-    val predictor = Trim.then(LowerCase())
-        .then(Tokenizer())
-        .then(new NGramsFeaturizer[String](1 to conf.nGrams))
-        .then(TermFrequency(x => 1))
-        .thenEstimator(CommonSparseFeatures(conf.commonFeatures))
-        .fit(trainData.data)
-        .thenLabelEstimator(NaiveBayesEstimator(numClasses))
-        .fit(trainData.data, trainData.labels)
-        .then(MaxClassifier)
+    val predictorPipeline = Trim andThen LowerCase() andThen
+        Tokenizer() andThen
+        NGramsFeaturizer(1 to conf.nGrams) andThen
+        TermFrequency(x => 1) andThen
+        (CommonSparseFeatures(conf.commonFeatures), trainData.data) andThen
+        (NaiveBayesEstimator(numClasses), trainData.data, trainData.labels) andThen
+        MaxClassifier
+
+
+    val predictor = Optimizer.execute(predictorPipeline)
```

Comment:
Does this mean we should add this to other pipelines as well now ? Or is it okay to not call the optimizer ? Also longer term we shouldn't expect the user to call this I guess, so we need to find a place to do this automatically